vng_api_common/authorizations/middleware.py

```diff
@@ -0,0 +1,230 @@
+from typing import Any, Dict, Iterable, List, Optional
+
+from django.conf import settings
+from django.db import models, transaction
+from django.db.models import QuerySet
+from django.utils.translation import gettext as _
+
+import jwt
+from djangorestframework_camel_case.util import underscoreize
+from rest_framework.exceptions import PermissionDenied
+from zds_client.client import ClientError
+
+from ..models import JWTSecret
+from ..utils import get_uuid_from_path
+from .models import Applicatie, AuthorizationsConfig, Autorisatie
+from .serializers import ApplicatieUuidSerializer
+
+
+class JWTAuth:
+    def __init__(self, encoded: str = None):
```

Comment:
Let's fix some of the type hints here as well

```suggestion
    def __init__(self, encoded: Optional[str] = None):
```